localstack/services/awslambda/lambda_executors.py

```diff
@@ -105,8 +105,11 @@ def get_main_endpoint_from_container():
             container_name = bootstrap.get_main_container_name()
             LOG.info('Unable to get IP address of main Docker container "%s": %s' %
                 (container_name, e))
-    # return (1) predefined endpoint host, or (2) main container IP, or (3) Docker host (e.g., bridge IP)
-    return config.HOSTNAME_FROM_LAMBDA or DOCKER_MAIN_CONTAINER_IP or config.DOCKER_HOST_FROM_CONTAINER
+    # return (1) Docker host (e.g., bridge IP), or (2) predefined endpoint host, or (3) main container IP
+    return (config.DOCKER_HOST_FROM_CONTAINER or
+    config.HOSTNAME_FROM_LAMBDA or
+    config.DOCKER_HOST_FROM_CONTAINER or
```

Comment:
Nitpick: I guess we can remove the duplicate `config.DOCKER_HOST_FROM_CONTAINER` here. 

Also, I think `config.HOSTNAME_FROM_LAMBDA` should take precedence and come first.

So, I guess the change here would be to change the order of (2) and (3):
```
return config.HOSTNAME_FROM_LAMBDA or config.DOCKER_HOST_FROM_CONTAINER or DOCKER_MAIN_CONTAINER_IP
```

Still - see my comment in the linked issue. I'm afraid that this change may actually break other people's configurations. Perhaps this is more of a documentation issue that we should solve via `localstack config validate`?

Reply:
100% agreed, as answered on the issue, this looks more like a config issue and how we want to manage this, I agree it can be easily solved via documentation.